From 11f6a85bb5f314ba8e1dbdc7654ef2647b75161e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jaime=20Marqu=C3=ADnez=20Ferr=C3=A1ndiz?=
 <jregistros+github@fastmail.net>
Date: Mon, 10 Nov 2025 15:42:49 +0100
Subject: [PATCH] fix: fix build with Qt 6.10 (#1728)

* Use the return value of QTemporaryFile::open. It is marked with [[nodiscard]] on Qt 6.10.
* Fix linking with Qt::GuiPrivate on Qt 6.10
---
 src/libs/ui/docsetsdialog.cpp                | 6 ++++--
 src/libs/ui/qxtglobalshortcut/CMakeLists.txt | 3 +++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/libs/ui/docsetsdialog.cpp b/src/libs/ui/docsetsdialog.cpp
index e2c936dc..8f2e86e1 100644
--- a/src/libs/ui/docsetsdialog.cpp
+++ b/src/libs/ui/docsetsdialog.cpp
@@ -342,7 +342,8 @@ void DocsetsDialog::downloadCompleted()
         QTemporaryFile *tmpFile = m_tmpFiles[docsetName];
         if (!tmpFile) {
             tmpFile = new QTemporaryFile(QStringLiteral("%1/%2.XXXXXX.tmp").arg(Core::Application::cacheLocation(), docsetName), this);
-            tmpFile->open();
+            if (!tmpFile->open())
+                return;
             m_tmpFiles.insert(docsetName, tmpFile);
         }
 
@@ -385,7 +386,8 @@ void DocsetsDialog::downloadProgress(qint64 received, qint64 total)
         QTemporaryFile *tmpFile = m_tmpFiles[docsetName];
         if (!tmpFile) {
             tmpFile = new QTemporaryFile(QStringLiteral("%1/%2.XXXXXX.tmp").arg(Core::Application::cacheLocation(), docsetName), this);
-            tmpFile->open();
+            if (!tmpFile->open())
+                return;
             m_tmpFiles.insert(docsetName, tmpFile);
         }
 
diff --git a/src/libs/ui/qxtglobalshortcut/CMakeLists.txt b/src/libs/ui/qxtglobalshortcut/CMakeLists.txt
index 5d029431..8a060663 100644
--- a/src/libs/ui/qxtglobalshortcut/CMakeLists.txt
+++ b/src/libs/ui/qxtglobalshortcut/CMakeLists.txt
@@ -42,6 +42,9 @@ elseif(UNIX AND X11_FOUND)
         find_package(Qt5 COMPONENTS X11Extras REQUIRED)
         target_link_libraries(QxtGlobalShortcut Qt5::X11Extras)
     else()
+        if(Qt6Core_VERSION VERSION_GREATER_EQUAL 6.10)
+            find_package(Qt6 COMPONENTS GuiPrivate REQUIRED)
+        endif()
         target_link_libraries(QxtGlobalShortcut Qt${QT_VERSION_MAJOR}::GuiPrivate)
     endif()
 
